oasislmf/manager.py

```diff
@@ -417,6 +418,67 @@ def generate_oasis_files(
             exposure_fp, _keys_fp, exposure_profile=exposure_profile
         )
 
+        # Fill summary of exposure data dictionary
+        def fill_exposure_summary(
+            exposure_summary_df,
+            exposure_summary,
+            peril,
+            status,
+        ):
+
+            for coverage_type in SUPPORTED_COVERAGE_TYPES:
+                tiv_sum = exposure_summary_df.loc[
+                    (exposure_summary_df['peril_id']==peril) &\
+                    (exposure_summary_df['coverage_type_id']==SUPPORTED_COVERAGE_TYPES[coverage_type]['id']), 'tiv'].sum()
+                tiv_sum = float(tiv_sum)
+                exposure_summary[peril][status]['tiv_by_coverage'][coverage_type] = tiv_sum
+                if coverage_type in exposure_summary[peril]['all']['tiv_by_coverage']:
+                    exposure_summary[peril]['all']['tiv_by_coverage'][coverage_type] += tiv_sum
+                else:
+                    exposure_summary[peril]['all']['tiv_by_coverage'][coverage_type] = tiv_sum
+                exposure_summary[peril][status]['tiv'] += tiv_sum
+                exposure_summary[peril]['all']['tiv'] += tiv_sum
+            exposure_summary[peril][status]['location_ids'] =\
+                exposure_summary_df.loc[
+                    exposure_summary_df['peril_id']==peril,
+                    'locnumber'].drop_duplicates().tolist()
+
+        # Get key errors in GUL input items dataframe format and merge with
+        # key errors to obtain status information
+        try:
+            gul_inputs_errors_df, _ = get_gul_input_items(
+                exposure_fp, _keys_errors_fp, exposure_profile=exposure_profile
+            )
+        except OasisException:
+            gul_inputs_errors_df = pd.DataFrame(columns=gul_inputs_df.columns)
+        keys_errors_df = pd.read_csv(_keys_errors_fp)
+        gul_inputs_errors_df = gul_inputs_errors_df.merge(
+            keys_errors_df,
+            left_on=['locnumber', 'peril_id', 'coverage_type_id'],
+            right_on=['LocID', 'PerilID', 'CoverageTypeID'],
+            how='left'
+        )
+        gul_inputs_errors_df = gul_inputs_errors_df.drop(
```

Comment:
Better when dropping columns (or rows) to use the `inplace=True` argument - that means the drop operations is performed on the object directly, and does not produce another object in memory which has to be reassigned.

Reply:
I think you also need `axis=1` here. So this line should be

    gul_inputs_errors_df.drop(columns=['locid', 'perilid', 'coveragetypeid', 'message'], axis=1, inplace=True)